crates/ruff_linter/src/rules/flake8_pyi/rules/custom_type_var_for_self.rs

```diff
@@ -359,68 +558,86 @@ fn import_self(checker: &Checker, position: TextSize) -> Result<(Edit, String),
         .get_or_import_symbol(&request, position, checker.semantic())
 }
 
-/// Returns a series of [`Edit`]s that modify all references to the given `typevar`,
-/// or `None` when it is not possible to resolve the binding.
+/// Returns a series of [`Edit`]s that modify all references to the given `typevar`.
 ///
 /// Only references within `editable_range` will be modified.
 /// This ensures that no edit in this series will overlap with other edits.
-fn replace_typevar_usages_with_self(
-    typevar: &ast::ExprName,
-    self_symbol_binding: &str,
+fn replace_typevar_usages_with_self<'a>(
+    typevar: &'a TypeVar<'a>,
+    self_or_cls_annotation_range: TextRange,
+    self_symbol_binding: &'a str,
     editable_range: TextRange,
-    semantic: &SemanticModel,
-) -> Option<Vec<Edit>> {
-    let binding = semantic
-        .only_binding(typevar)
-        .map(|id| semantic.binding(id))?;
-
-    let mut edits = vec![];
-
-    for reference_id in binding.references() {
-        let reference = semantic.reference(reference_id);
-        let range = reference.range();
-
-        if editable_range.contains_range(range) {
-            let edit = Edit::range_replacement(self_symbol_binding.to_string(), range);
-            edits.push(edit);
-        }
-    }
-
-    Some(edits)
+    semantic: &'a SemanticModel<'a>,
+) -> impl Iterator<Item = Edit> + 'a {
+    typevar
+        .references(semantic)
+        .map(Ranged::range)
+        .filter(move |reference_range| editable_range.contains_range(*reference_range))
+        .filter(move |reference_range| {
+            !self_or_cls_annotation_range.contains_range(*reference_range)
+        })
+        .map(|reference_range| {
+            Edit::range_replacement(self_symbol_binding.to_string(), reference_range)
+        })
 }
 
-fn remove_typevar_declaration(type_params: Option<&ast::TypeParams>, name: &str) -> Option<Edit> {
-    let is_declaration_in_question = |type_param: &&ast::TypeParam| -> bool {
-        if let ast::TypeParam::TypeVar(typevar) = type_param {
-            return typevar.name.as_str() == name;
-        };
-
-        false
-    };
-
-    let parameter_list = type_params?;
-    let parameters = &parameter_list.type_params;
-    let first = parameters.first()?;
-
-    if parameter_list.len() == 1 && is_declaration_in_question(&first) {
-        return Some(Edit::range_deletion(parameter_list.range));
+/// Create an [`Edit`] removing the `TypeVar` binding from the PEP 695 type parameter list.
+///
+/// Return `None` if we fail to find a `TypeVar` that matches the range of `typevar_binding`.
+fn remove_pep695_typevar_declaration(
+    type_params: &ast::TypeParams,
+    custom_typevar: &TypeVar,
+) -> Option<Edit> {
+    let parameters = &type_params.type_params;
+
+    if type_params.len() == 1 {
+        let sole_typevar = &parameters[0];
+        return (sole_typevar.range() == custom_typevar.range())
+            .then(|| Edit::range_deletion(type_params.range));
```

Comment:
```suggestion
```suggestion
    if let [sole_typevar] == parameters.as_slice() {
        return (sole_typevar.range() == custom_typevar.range())
            .then(|| Edit::range_deletion(type_params.range));
    }
```